File: src/main/scala/org/openapitools/client/api/ParcelBoxLabelDTO.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ParcelBoxLabelDTO._

case class ParcelBoxLabelDTO (
  /* Соответствует URL, по которому выполняется запрос [GET campaigns/{campaignId}/orders/{orderId}/delivery/shipments/{shipmentId}/boxes/{boxId}/label](../../reference/orders/generateOrderLabel.md).  */
  url: String,
/* Юридическое название магазина. */
  supplierName: String,
/* Юридическое название службы доставки. */
  deliveryServiceName: String,
/* Идентификатор заказа в системе Маркета. */
  orderId: Long,
/* Идентификатор заказа в информационной системе магазина.  Совпадает с `orderId`, если Маркету неизвестен номер заказа в системе магазина.  */
  orderNum: String,
/* Фамилия и инициалы получателя заказа. */
  recipientName: String,
/* Идентификатор коробки. */
  boxId: Long,
/* Идентификатор коробки в информационной системе магазина.  Возвращается в формате: `номер заказа на Маркете-номер коробки`. Например, `7206821‑1`, `7206821‑2` и т. д.  */
  fulfilmentId: String,
/* Номер коробки в заказе. Возвращается в формате: `номер места/общее количество мест`.  */
  place: String,
/* {% note warning \"\" %}  Этот параметр устарел. Не используйте его.  {% endnote %}  Общая масса всех товаров в заказе. Возвращается в формате: `weight кг`.  */
  weight: String,
/* Идентификатор службы доставки. Информацию о службе доставки можно получить с помощью запроса [GET delivery/services](../../reference/orders/getDeliveryServices.md). */
  deliveryServiceId: String,
/* Адрес получателя. */
  deliveryAddress: Option[String],
/* Дата отгрузки в формате `dd.MM.yyyy`. */
  shipmentDate: Option[String])

object ParcelBoxLabelDTO {
  import DateTimeCodecs._

  implicit val ParcelBoxLabelDTOCodecJson: CodecJson[ParcelBoxLabelDTO] = CodecJson.derive[ParcelBoxLabelDTO]
  implicit val ParcelBoxLabelDTODecoder: EntityDecoder[ParcelBoxLabelDTO] = jsonOf[ParcelBoxLabelDTO]
  implicit val ParcelBoxLabelDTOEncoder: EntityEncoder[ParcelBoxLabelDTO] = jsonEncoderOf[ParcelBoxLabelDTO]
}

File: API/Client/ParcelBoxLabelDTO.js
goog.provide('API.Client.ParcelBoxLabelDTO');

/**
 * Информация о ярлыке для коробки.
 * @record
 */
API.Client.ParcelBoxLabelDTO = function() {}

/**
 * Соответствует URL, по которому выполняется запрос [GET campaigns/{campaignId}/orders/{orderId}/delivery/shipments/{shipmentId}/boxes/{boxId}/label](../../reference/orders/generateOrderLabel.md). 
 * @type {!string}
 * @export
 */
API.Client.ParcelBoxLabelDTO.prototype.url;

/**
 * Юридическое название магазина.
 * @type {!string}
 * @export
 */
API.Client.ParcelBoxLabelDTO.prototype.supplierName;

/**
 * Юридическое название службы доставки.
 * @type {!string}
 * @export
 */
API.Client.ParcelBoxLabelDTO.prototype.deliveryServiceName;

/**
 * Идентификатор заказа в системе Маркета.
 * @type {!number}
 * @export
 */
API.Client.ParcelBoxLabelDTO.prototype.orderId;

/**
 * Идентификатор заказа в информационной системе магазина.  Совпадает с `orderId`, если Маркету неизвестен номер заказа в системе магазина. 
 * @type {!string}
 * @export
 */
API.Client.ParcelBoxLabelDTO.prototype.orderNum;

/**
 * Фамилия и инициалы получателя заказа.
 * @type {!string}
 * @export
 */
API.Client.ParcelBoxLabelDTO.prototype.recipientName;

/**
 * Идентификатор коробки.
 * @type {!number}
 * @export
 */
API.Client.ParcelBoxLabelDTO.prototype.boxId;

/**
 * Идентификатор коробки в информационной системе магазина.  Возвращается в формате: `номер заказа на Маркете-номер коробки`. Например, `7206821‑1`, `7206821‑2` и т. д. 
 * @type {!string}
 * @export
 */
API.Client.ParcelBoxLabelDTO.prototype.fulfilmentId;

/**
 * Номер коробки в заказе. Возвращается в формате: `номер места/общее количество мест`. 
 * @type {!string}
 * @export
 */
API.Client.ParcelBoxLabelDTO.prototype.place;

/**
 * {% note warning \"\" %}  Этот параметр устарел. Не используйте его.  {% endnote %}  Общая масса всех товаров в заказе. Возвращается в формате: `weight кг`. 
 * @type {!string}
 * @export
 */
API.Client.ParcelBoxLabelDTO.prototype.weight;

/**
 * Идентификатор службы доставки. Информацию о службе доставки можно получить с помощью запроса [GET delivery/services](../../reference/orders/getDeliveryServices.md).
 * @type {!string}
 * @export
 */
API.Client.ParcelBoxLabelDTO.prototype.deliveryServiceId;

/**
 * Адрес получателя.
 * @type {!string}
 * @export
 */
API.Client.ParcelBoxLabelDTO.prototype.deliveryAddress;

/**
 * Дата отгрузки в формате `dd.MM.yyyy`.
 * @type {!string}
 * @export
 */
API.Client.ParcelBoxLabelDTO.prototype.shipmentDate;

